sbx/common/jax_layers.py

```diff
@@ -0,0 +1,206 @@
+from typing import Any, Callable, Optional, Sequence, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+from flax.linen.module import Module, compact, merge_param
+from flax.linen.normalization import _canonicalize_axes, _compute_stats, _normalize
+from jax.nn import initializers
+
+PRNGKey = Any
+Array = Any
+Shape = Tuple[int, ...]
+Dtype = Any  # this could be a real type?
+Axes = Union[int, Sequence[int]]
```

Comment:
Flax v0.8.1 introduced [flax.typing](https://github.com/google/flax/pull/3624), which we could use here for more descriptive type hints, similar to the current version of [flax.linen.normalization](https://flax.readthedocs.io/en/latest/_modules/flax/linen/normalization.html). However, we should probably wait a bit here since this would require a relatively recent flax version.